convert2rhel/unit_tests/toolopts_test.py

```diff
@@ -145,3 +146,254 @@ def test_both_disable_submgr_and_no_rhsm_options_work(argv, raise_exception, no_
+        ),
+        (
+            mock_cli_arguments(["-k", "activation_key"]),
+            "[subscription_manager]\nactivation_key=conf_key",
+            {"password": None, "activation_key": "activation_key"},
+            "Command line authentication method take precedence over method in configuration file.",
+        ),
+        (
+            mock_cli_arguments(["-k", "activation_key"]),
+            "[subscription_manager]\npassword=conf_pass",
+            {"password": "conf_pass", "activation_key": "activation_key"},
+            "Command line authentication method take precedence over method in configuration file.",
+        ),
+        (
+            mock_cli_arguments(["-k", "activation_key", "-p", "password"]),
+            "[subscription_manager]\npassword=conf_pass\nactivation_key=conf_key",
+            {"password": "password", "activation_key": "activation_key"},
+            "Command line authentication method take precedence over method in configuration file.",
+        ),
+        (
+            mock_cli_arguments([""]),
+            "[subscription_manager]\npassword=conf_pass\nactivation_key=conf_key",
+            {"password": "conf_pass", "activation_key": "conf_key"},
+            "Set only one of password or activation key. Activation key take precedence.",
+        ),
+    ),
+)
+def test_config_file(argv, content, output, message, monkeypatch, tmp_path, caplog):
+    # After each test there were left data from previous
+    # Re-init needed delete the set data
+    convert2rhel.toolopts.tool_opts.__init__()
+    path = os.path.join(str(tmp_path), "convert2rhel.ini")
+    with open(path, "w") as file:  # pylint: disable=unspecified-encoding
+        file.write(content)
+    os.chmod(path, 0o600)
+
+    monkeypatch.setattr(sys, "argv", argv)
+    monkeypatch.setattr(convert2rhel.toolopts, "CONFIG_PATHS", value=[path])
+    convert2rhel.toolopts.CLI()
+
+    assert convert2rhel.toolopts.tool_opts.activation_key == output["activation_key"]
+    assert convert2rhel.toolopts.tool_opts.password == output["password"]
+    if message:
+        assert message in caplog.text
+
+
+@pytest.mark.parametrize(
+    ("argv", "content", "message", "output"),
+    (
+        (
+            mock_cli_arguments(["--password", "pass", "-f"]),
+            "pass_file",
+            "Password file argument take precedence over the password argument.",
+            {"password": "pass_file", "activation_key": None},
+        ),
+        (
+            mock_cli_arguments(["--password", "pass", "--config-file"]),
+            "[subscription_manager]\nactivation_key = key_cnf_file",
+            "Command line authentication method take precedence over method in configuration file.",
+            {"password": "pass", "activation_key": None},
+        ),
+    ),
+)
+def test_multiple_auth_src_combined(argv, content, message, output, caplog, monkeypatch, tmp_path):
+    """Test combination of password file or configuration file and CLI arguments."""
```

Comment:
I would say we don't need to have docstrings in the test functions, in this case, it is pretty clear to me what the test is about. 
```suggestion
```

Reply:
@r0x0d Yeah, they are maybe a bit unnecessary, but they helped me a lot while I was creating the test to keep me organized. If they should be removed, then it should be the last step :smile: Or they could help somehow in the future.